spec/classes/profile/tsm_spec.rb

```diff
@@ -61,13 +62,19 @@
             domains: ['/baz', '/quux'],
             virtualmountpoints: ['/vmount'],
             exclude_dirs: ['/whatever'],
+            opt_settings: [
+              'first_setting first_value',
+              'second_setting "second_value"',
+            ],
           )
         end
 
         it do
           is_expected.to contain_file(dsm_opt)
             .with_content(%r{^DOMAIN "/baz"$})
             .with_content(%r{^DOMAIN "/quux"$})
+            .with_content(%r{^first_setting first_value$})
+            .with_content(%r{^second_setting "second_value"$})
         end
```

Comment:
I'd consider splitting this into two tests, as it's testing two different behaviors (domains and opt_settings). For example, if you had

```ruby
it "adds domain settings to the dsm config file" do
  is_expected.to contain_file(dsm_opt)
    .with_content(%r{^DOMAIN "/baz"$})
    .with_content(%r{^DOMAIN "/quux"$})
end

it "adds custom settings to the dsm config file" do
  is_expected.to contain_file(dsm_opt)
    .with_content(%r{^first_setting first_value$})
    .with_content(%r{^second_setting second_value$})
end
```

The advantage of doing it this way is that the errors will be clearer in the future if one of these tests fails. It also shows that, while they look similar, they serve different purposes. If at a later date we ever wanted to change the domain behavior, for example, we wouldn't have to untangle which parts of the test test what.

Not telling you to do this, by the way; I think much of the usefulness is just in considering it. Writing good tests is always a complex web of trade-offs.

Reply:
I do appreciate the info. It helps in learning and getting better. I can split it up; makes sense to do.